examples/jsm/csm/CSM.js

```diff
@@ -200,14 +204,20 @@ export class CSM {
 
 		const camera = this.camera;
 		const frustums = this.frustums;
+
+		// for each frustum we need to find its min-max box aligned with the light orientation
+		// the position in _lightOrientationMatrix does not matter, as we transform there and back
+		_lightTarget.copy( this.lightDirection ).add( camera.position );
+		_lightOrientationMatrix.lookAt( camera.position, _lightTarget, _up );
+		_lightOrientationMatrixInverse.copy(_lightOrientationMatrix).invert();
+
 		for ( let i = 0; i < frustums.length; i ++ ) {
 
 			const light = this.lights[ i ];
 			const shadowCam = light.shadow.camera;
 			const texelWidth = ( shadowCam.right - shadowCam.left ) / this.shadowMapSize;
 			const texelHeight = ( shadowCam.top - shadowCam.bottom ) / this.shadowMapSize;
-			light.shadow.camera.updateMatrixWorld( true );
```

Comment:
Wouldn't it be possible to use the following code block instead:
```js
light.updateWorldMatrix( true, false );
light.target.updateWorldMatrix( true, false );

light.shadow.updateMatrices( light );
```
Shouldn't this code also ensure the shadow camera has a correct world matrix for further computation?

Reply:
That is the chicken-egg problem. Light position is not set yet, it will be set later in this function based on the frustum bounding box.